linkerd/client-policy/src/lib.rs

```diff
@@ -0,0 +1,146 @@
+#![deny(rust_2018_idioms, clippy::disallowed_methods, clippy::disallowed_types)]
+#![forbid(unsafe_code)]
+
+pub mod http;
+
+#[cfg(feature = "proto")]
+pub mod proto;
+
+use linkerd_addr::{Addr, NameAddr};
+// pub use linkerd_policy_core::{meta, Meta};
+use linkerd_proxy_api_resolve as resolve;
+use std::{fmt, net::SocketAddr, str::FromStr, sync::Arc};
+use tokio::sync::watch;
+
+type Backends = Arc<[Backend]>;
+
+#[derive(Clone, Debug)]
+pub struct Receiver {
+    inner: watch::Receiver<Policy>,
+}
+
+// TODO(eliza): this is looking a lot like the `Profile` type...could be made
+// generic eventually...
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct Policy {
+    pub addr: Option<LogicalAddr>,
+    pub http_routes: Arc<[http::Route]>,
+    pub opaque_protocol: bool,
+    pub endpoint: Option<(SocketAddr, resolve::Metadata)>,
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+pub struct RoutePolicy {
+    pub backends: Backends,
+    // pub meta: Arc<Meta>,
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+pub struct Backend {
+    pub weight: u32,
+    pub addr: Addr,
+}
+
+/// A profile lookup target.
+#[derive(Clone, Hash, Eq, PartialEq)]
+pub struct LookupAddr(pub Addr);
+
+/// A bound logical service address
+#[derive(Clone, Hash, Eq, PartialEq)]
+pub struct LogicalAddr(pub NameAddr);
```

Comment:
We should be using Meta/ref types for anything related to addrs in policy-land

Reply:
the logical address is coming from the protobuf's `backend` field; turning that into a meta type will need a change in proxy-api as well.

Reply:
Yep. ~Everything about the proxy api draft is on the table to change :) This is why I want to work 'out' from the logical stacks -- let the proxy's needs dictate the API and not vice-versa.

Reply:
yeah, that makes sense.